### DISCUSSION TITLE
forge test keep redeploying mock contract on anvil

### DISCUSSION THREAD

**Discussion by Stachelabs** [Cyfrin/foundry-full-course-cu#2767] [Q&A]
Ola, I'm using Patrick's script in order to deploy a mock contract on Anvil using:

```solidity
function getOrCreateAnvilEthConfig() public returns (NetworkConfig memory) {
    // Check to see if we set an active network config
    // Won't redeploy the mock if we already have a mock price feed

    if (localNetworkConfig.priceFeed != address(0)) {
        return localNetworkConfig;
    }

    console2.log(unicode"⚠️ You have deployed a mock contract!");
    console2.log("Make sure this was intentional");
    vm.startBroadcast();
    MockV3Aggregator mockPriceFeed = new MockV3Aggregator(
        DECIMALS,
        INITIAL_PRICE
    );
    vm.stopBroadcast();

    localNetworkConfig = NetworkConfig({priceFeed: address(mockPriceFeed)});
    return localNetworkConfig;
}
```

However, in my thinking, once one mock is deployed, there should be no need to redeploy another mock, but the code keeps redeploying a mock when using:

```bash
forge test --rpc-url $ANVIL_RPC_URL -vv
```

Is it normal that the test cleans and re-runs the setup config at every test?

Thank you! 🙂

**Comment by EngrPips** [comment]
The test runs a new process every time you run a test. So I think that is completely normal.

**Reply by Stachelabs** [reply]
thank you that's what I was thinking, it might be useful if the script is use somewhere else than in testing environment? wdyt?

**Reply by EngrPips** [reply]
Yeah you can play with stuff.
